src/includes/class-wordlift-post-to-jsonld-converter.php

```diff
@@ -303,4 +317,74 @@ protected function set_publisher_by_reference( &$jsonld, &$references ) {
 		$references[] = $publisher_id;
 	}
 
+	/**
+	 * Add the extra fields to the Publisher JSON-LD structure.
+	 *
+	 * @param &$publisher_jsonld array Reference to the Publisher JSON-LD array within the main JSON-LD array.
+	 * @param $post_id           int   The current Post ID.
+	 *
+	 * @since 3.53.0
+	 */
+	public function expand_publisher_jsonld( &$publisher_jsonld, $post_id ) {
+		$publisher_id = Wordlift_Configuration_Service::get_instance()->get_publisher_id();
+
+		// No publisher set. Exit.
+		if ( ! isset( $publisher_id ) || ! is_numeric( $publisher_id ) ) {
+			return;
+		}
+
+		// Exit if current Post is not Publisher post
+		if ( $post_id !== $publisher_id ) {
+			return;
+		}
+
+		// Add alternativeName if set.
+		$alternative_name = $this->configuration_service->get_alternate_name();
+
+		if ( ! empty( $alternative_name ) ) {
+			$publisher_jsonld['alternateName'] = $alternative_name;
+		}
+
+		/*
+		 * Set the logo, only for http://schema.org/ + Organization, LocalBusiness, or OnlineBusiness
+		 * as Person doesn't support the logo property.
+		 *
+		 * @see http://schema.org/logo.
+		 */
+		$organization_types = array(
+			'Organization',
+			'LocalBusiness',
+			'OnlineBusiness',
+		);
+
+		if ( ! in_array( $publisher_jsonld['@type'], $organization_types, true ) ) {
+			return;
+		}
+
+		// Get the publisher logo.
+		$publisher_logo = $this->publisher_service->get_publisher_logo( $publisher_id );
```

Comment:
This part must go in the entity-to-jsonld converter, we set the publisher logo if the post being converted is the publisher.